### PR TITLE
small fixes and defining publication view 

### PR DESCRIPTION
**Pull request by vasgat**

### CODE REVIEW

File: mod/wikirate_project/data/real.yml [hunk 1]
```diff
@@ -1054,6 +1059,18 @@
...
+- name: "Events Description"
+  type: :basic
+  codename: :events_description
+  content: "Here you can find all upcoming and past events."
+
+- :name: "Image on Socials"
+  :type: :image
+  :codename: :image_on_socials
+  :image: image_on_socials.png
+  :storage_type: :coded
```
Comment (ethn) [inline mod/wikirate_project/data/real.yml:1071]: fyi, in concept `coded` images are supposed to be for images bound to the mod code. As in, the image should be stored in the code, not in the cloud.  I'm not convinced it's all working properly, but you can probably skip that here.

File: mod/wikirate_project/set/self/about_us.rb [hunk 2]
```diff
@@ -7,4 +7,12 @@
...
   end
+
+  def social_image
+    Card[:image_on_socials]&.format(:text)&.render_source
+  end
+
+  def social_description
+    Card.fetch(:social_description)&.format(:text)&.text_description
+  end
 end
```
Comment (ethn) [inline mod/wikirate_project/set/self/about_us.rb:18]: For the repeated code (the `social_image`), the preferred method is to make an abstract set.  Eg abstract/social_image.rb.  Then you use `include_set Abstract::SocialImage` in these files.

For the descriptions, is there a reason not to use the default behavior and create an `:about_us+:description` card? If that's already taken, you could generalize this to use a different field in the abstract mod I just mentioned.

**Review by ethn — CHANGES_REQUESTED** [review]
I would suggest a little less duplication in the social media fields (see comments). Otherwise looks great!

File: mod/wikirate_project/data/real.yml (hunk 1 above)
Comment (vasgat) [reply mod/wikirate_project/data/real.yml]: I removed the storage_type field from the card

File: mod/wikirate_project/set/self/about_us.rb (hunk 2 above)
Comment (vasgat) [reply mod/wikirate_project/set/self/about_us.rb:12]: I have addressed both your comments on the following commits.

**Pull request merged**